### PR TITLE
Cecilia/paddle reader

### PR DESCRIPTION
**Pull request by ceciliapeng2011**
Refactor python API read_network, and add test case for it.

1. update comment description to add paddle model support.
2. remove assumption of file directory is not allowed. in case of paddle, as well as tensorflow, a kind of scattered model representation is possible. That is, a model file as well as a group of parameter files for each of them are weights of each layer.

note: the test case has a dependency to -
https://github.com/openvinotoolkit/testdata/pull/6

### CODE REVIEW

File: inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api.pyx [hunk 1]
```diff
@@ -312,9 +312,9 @@ cdef class IECore:
         return versions
 
     ## Reads a network from Intermediate Representation (IR) or ONNX formats and creates an `IENetwork`.
-    #  @param model: A `.xml` or `.onnx` model file or string with IR.
+    #  @param model: A `.xml` or `.onnx` or `.pdmodel` model file or string with IR.
```
Comment (AlexeyLebedev1) [inline inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api.pyx:315]:
```suggestion
    #  @param model: A `.xml`,  `.onnx` or `.pdmodel` model file or string with IR.
```

**Review by AlexeyLebedev1 — CHANGES_REQUESTED** [review]

**Comment by ceciliapeng2011** [comment]
temporarily converted to Draft, as it has dependency PR to testdata.

File: inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api.pyx [hunk 2]
```diff
@@ -340,8 +340,6 @@ cdef class IECore:
         else:
             weights_ = "".encode()
             model = os.fspath(model)
-            if not os.path.isfile(model):
```
Comment (mvafin) [inline inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api.pyx:343]: I suggest changing this from the requirement for a path to be a file path to just checking the existence of the path with `os.path.exists()`

**Comment by ceciliapeng2011** [comment]
Dependency to testdata merged. So folks, please continue this PR review. Thanks! @mvafin @AlexeyLebedev1

**Pull request closed without merging**